plugin/trino-iceberg/src/test/java/io/trino/plugin/iceberg/catalog/glue/TestIcebergGlueCatalogConnectorSmokeTest.java

```diff
@@ -134,6 +146,46 @@ public void testRenameSchema()
                 .hasStackTraceContaining("renameNamespace is not supported for Iceberg Glue catalogs");
     }
 
+    @Test
+    public void testShowTablesInAccessDeniedSchema()
+    {
+        String schemaName = "test_permission_denied_" + randomTableSuffix();
+        assertUpdate("CREATE SCHEMA " + schemaName);
+
+        try (TestTable table = new TestTable(getQueryRunner()::execute, schemaName + ".test_show_tables", "(col int)")) {
+            String tableName = table.getName().substring(schemaName.length() + 1);
+            Table glueTable = glueClient.getTable(new GetTableRequest().withDatabaseName(schemaName).withName(tableName)).getTable();
+            String user = glueTable.getCreatedBy();
+            String accountId = Arn.fromString(user).getAccountId();
+            String policy = """
+                    {
+                        "Statement": [
+                            {
+                                "Principal": {
+                                    "AWS": [
+                                        "%s"
+                                    ]
+                                },
+                                "Sid": "GetTablesAccessDeny",
+                                "Effect": "Deny",
+                                "Action": [
+                                    "glue:GetTables"
+                                ],
+                                "Resource": [
+                                    "arn:aws:glue:%s:%s:database/%s"
+                                ]
+                            }
+                        ]
+                    }
+                    """.formatted(user, region, accountId, schemaName);
+            glueClient.putResourcePolicy(new PutResourcePolicyRequest().withPolicyInJson(policy));
+            assertQueryReturnsEmptyResult("SHOW TABLES IN " + schemaName);
+        }
+        finally {
+            glueClient.deleteDatabase(new DeleteDatabaseRequest().withName(schemaName));
```

Comment:
Does the dynamically added resource policy disappear once that the database is being deleted?

Reply:
Added code to delete the resource policy explicitly.